pkg/logs/auditor/auditor.go

```diff
@@ -55,33 +56,47 @@ type Auditor struct {
 }
 
 // New returns an initialized Auditor
-func New(inputChan chan *message.Message, runPath string) *Auditor {
+func New(runPath string) *Auditor {
 	return &Auditor{
-		inputChan:    inputChan,
 		registryPath: filepath.Join(runPath, "registry.json"),
 		entryTTL:     defaultTTL,
-		done:         make(chan struct{}),
 	}
 }
 
 // Start starts the Auditor
 func (a *Auditor) Start() {
+	a.inputChan = make(chan *message.Message, config.ChanSize)
+	a.done = make(chan struct{})
 	a.registry = a.recoverRegistry()
 	a.cleanupRegistry()
 	go a.run()
 }
 
 // Stop stops the Auditor
 func (a *Auditor) Stop() {
-	close(a.inputChan)
-	<-a.done
+	if a.inputChan != nil {
+		close(a.inputChan)
+	}
+
+	if a.done != nil {
+		<-a.done
+		a.done = nil
+	}
+	a.inputChan = nil
+
 	a.cleanupRegistry()
 	err := a.flushRegistry()
 	if err != nil {
 		log.Warn(err)
 	}
 }
 
+// Channel returns the channel to use to communicate with the auditor or nil
+// if the auditor is currently stopped.
+func (a *Auditor) Channel() chan *message.Message {
```

Comment:
The channel could also be public, see the go ticker for example
https://golang.org/pkg/time/#Ticker

Reply:
I know, I though it would be cleaner to have a stupid accessor.